### PR TITLE
Add extra stack definitions

### PR DESCRIPTION
**Pull request by duncangrant**
Accepts urls to tar.gz, tar, and zip archives
Urls can be http(s), classpath, file

### CODE REVIEW

File: ambari/src/main/java/io/brooklyn/ambari/AmbariClusterImpl.java [hunk 1]
```diff
@@ -179,6 +190,11 @@ public void start(Collection<? extends Location> locations) {
         return Entities.descendants(this, AmbariServer.class);
     }
 
+    @Override
+    public List<String> getExtraStackDefinitionsUrls() {
+        return this != null ? getConfig(STACK_DEFINITION_URLS) : ImmutableList.<String>of();
```
Comment (tbouron) [inline ambari/src/main/java/io/brooklyn/ambari/AmbariClusterImpl.java]: Shouldn't be more like `return getConfig(STACK_DEFINITION_URLS);` ?

`this` is always different than `null` as you call this method, and `getConfig(STACK_DEFINITION_URLS)` is set with a default empty list.

**Comment by tbouron** [comment]
You shouldn't have remove `service/src/main/resources/archetype-resources/src/main/java/MyService.java` and `service/src/main/resources/archetype-resources/src/main/java/MyServiceImpl.java`.

Also, the `ambari/src/test/resources/nslcd-stack.tar.gz` shouldn't be hosted onto another github repo?

**Comment by duncangrant** [comment]
you're correct - I'll fix those

**Comment by tbouron** [comment]
LGTM, merging.

**Pull request merged**